nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -23,6 +23,45 @@ let
     catch_workers_output = "yes";
   };
 
+  appStores = {
+    # default apps bundled with pkgs.nextcloudXX, e.g. files, contacts
+    apps = {
+      enabled = true;
+      writable = false;
+    };
+    # apps installed via cfg.extraApps
+    nix-apps = {
+      enabled = cfg.extraApps != { };
+      linkTarget = "${pkgs.linkFarm "nix-apps"
+        (mapAttrsToList (name: path: { inherit name path; }) cfg.extraApps)}";
+      writable = false;
+    };
+    # apps installed via the app store.
+    store-apps = {
+      enabled = cfg.appstoreEnable == null || cfg.appstoreEnable;
+      linkTarget = "${cfg.home}/store-apps";
+      writable = true;
+    };
+  };
+
+  webroot = pkgs.runCommand
+    "${cfg.package.name or "nextcloud"}-with-apps"
+    { nativeBuildInputs = [ pkgs.xorg.lndir ]; }
+    ''
+      mkdir $out
+      pushd "$out" &>/dev/null
+        lndir "${cfg.package}"
```

Comment:
lndir also supports a target directory, so we could avoid the pushd/popd.

Also do we really need to use lndir or would ln -s be enough?

Reply:
Good point, done with `ln -s`!